api/placer.py

```diff
@@ -138,7 +138,8 @@ def check(self):
         metadata_validator = validators.from_schema_path(payload_schema_uri)
         metadata_validator(self.metadata, 'POST')
 
-        targets = self.create_hierarchy(self.metadata)
+        # If not a superuser request, pass uid of user making the upload request
```

Comment:
Possibly .. it's hard because before this point, we only know of the uid or label and `hierarchy.py` does a lot of work to look these item up by this information as well as fuzzy match on group/project. If we wait until after running `create_hierarchy`, the hierarchy has already been created/modified. 

We could maybe expose some of the lookup functions?